2024-RIKEN-AWS/JupyterNotebook/docker/Dockerfile.spawn

```diff
@@ -9,32 +9,51 @@ USER root
 ENV NB_USER=jovyan \
     NB_UID=1000 \
     HOME=/home/jovyan
+    # VENV_DIR=/home/jovyan/.flux_tutorial_venv
 
 RUN adduser \
-        --disabled-password \
-        --gecos "Default user" \
-        --uid ${NB_UID} \
-        --home ${HOME} \
-        --force-badname \
-        ${NB_USER}
+    --disabled-password \
+    --gecos "Default user" \
+    --uid ${NB_UID} \
+    --home ${HOME} \
+    --force-badname \
+    ${NB_USER}
 
 RUN apt-get update \
- && apt-get upgrade -y \
- && apt-get install -y --no-install-recommends \
-        ca-certificates \
-        dnsutils \
-        iputils-ping \
-        python3-pip \
-        tini \
-        # requirement for nbgitpuller
-        git \
- && rm -rf /var/lib/apt/lists/*
+    # && apt-get upgrade -y \
+    && apt-get install -y --no-install-recommends \
+    gcc-10 \
+    g++-10 \
+    ca-certificates \
+    dnsutils \
+    iputils-ping \
+    python3.9 \
+    python3.9-dev \
+    python3-pip \
+    python3-venv \
+    openmpi-bin \
+    openmpi-common \
+    libopenmpi-dev \
+    liblz4-dev \
+    tini \
+    # requirement for nbgitpuller
+    git
+#&& rm -rf /var/lib/apt/lists/*
```

Comment:
Can this be uncommented to clean up extra files?

Reply:
Probably, although I haven't tested it since we got DLIO working. I can always uncomment it, and let the CI take a crack at building the image. If it works for the CI, it works for me

Reply:
The build on CI worked with this uncommented so it should be fine.